packages/ember-glimmer/tests/integration/input-test.js

```diff
@@ -0,0 +1,114 @@
+import { RenderingTest, moduleFor } from '../utils/test-case';
+import { set } from 'ember-metal/property_set';
+
+const camelizeMap = {
+  tabindex: 'tabIndex',
+  maxlength: 'maxLength'
+};
+
+moduleFor('Input element tests', class extends RenderingTest {
+
+  runAttributeTest(attributeName, values) {
+    let template = `<input ${attributeName}={{value}}>`;
+    this.render(template, { value: values[0] });
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} is set on initial render`);
+
+    this.runTask(() => this.rerender());
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} is set on noop rerender`);
+
+    this.setComponentValue(values[1]);
+    this.assertAttributeHasValue(attributeName, values[1], `${attributeName} is set on rerender`);
+
+    this.setComponentValue(values[0]);
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} can be set back to the initial value`);
+  }
+
+  ['@test input disabled attribute']() {
+    this.runAttributeTest('disabled', [false, true]);
+  }
+
+  ['@test input value attribute']() {
+    this.runAttributeTest('value', ['foo', 'bar']);
+  }
+
+  ['@test input placeholder attribute']() {
+    this.runAttributeTest('placeholder', ['placeholder', 'facebolder']);
+  }
+
+  ['@test input name attribute']() {
+    this.runAttributeTest('name', ['nam', 'name']);
+  }
+
+  ['@htmlbars input maxlength attribute']() {
+    this.runAttributeTest('maxlength', [2, 3]);
+  }
+
+  ['@test input size attribute']() {
+    this.runAttributeTest('size', [2, 3]);
+  }
+
+  ['@htmlbars input tabindex attribute']() {
+    this.runAttributeTest('tabindex', [2, 3]);
+  }
+
+  ['@test cursor position is not lost when updating content']() {
+    let template = `<input value={{value}}>`;
+    this.render(template, { value: 'hola' });
+
+    this.setDOMValue('hola');
+    this.setSelectionRange(1, 3);
+
+    this.setComponentValue('hola');
```

Comment:
and here you'll want to `this.setComponentValue('hello')` to ensure that it is a no-op against the DOM's current state rather than the "last set value".